timemachine/fe/single_topology.py

```diff
@@ -1088,6 +1146,71 @@ def assert_chiral_consistency(src_chiral_idxs: NDArray, dst_chiral_idxs: NDArray
     assert len(dst_chiral_restr_idx_set.allowed_set.intersection(src_chiral_restr_idx_set.disallowed_set)) == 0
 
 
+@dataclass
+class AlignedPotential:
+    idxs: NDArray[np.int32]
+    src_params: NDArray[np.float64]
+    dst_params: NDArray[np.float64]
+    mins: NDArray[np.float64]
+    maxes: NDArray[np.float64]
+
+    def interpolate(self, lamb):
+        raise NotImplementedError()
+
+
+class AlignedBond(AlignedPotential):
+    def interpolate(self, lamb):
+        k_min = 0.1
+        params = batch_interpolate_harmonic_bond_params(
+            self.src_params, self.dst_params, lamb, k_min, self.mins, self.maxes
+        )
+        params = jnp.array(params, dtype=np.float64).T
```

Comment:
nit: is this an unnecessary copy? (`params` is already a jax array if I'm reading correctly)

Reply:
`params` is return from `vmap` around `interpolate_harmonic_bond_params`, which returns `lists`